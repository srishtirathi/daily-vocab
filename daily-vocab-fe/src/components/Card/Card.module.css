.card {
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
    transition: 0.3s;
    width: 50rem;
    border-radius: 5px;
    background-color: cadetblue;
    margin: 1rem;
}

.container {
    padding: 2px 16px;
}