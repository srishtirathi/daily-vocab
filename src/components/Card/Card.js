import React from 'react';
import PropTypes from 'prop-types';
import styles from './Card.module.css';

const Card = ({ vocab }) => (
  <div className={styles.card}>

    <div className={styles.container}>

      <h4><b>{vocab.word}</b></h4>
      <p>{vocab.sentence}</p>
    </div>

  </div>
);

Card.propTypes = {

  vocab: PropTypes.shape({
    word: PropTypes.string.isRequired,
    sentence: PropTypes.string.isRequired,

  }).isRequired,
};

export default Card;
