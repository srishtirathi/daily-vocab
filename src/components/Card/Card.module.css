.card {
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
    transition: 0.3s;
    width: 40%;
    border-radius: 5px;
    background-color: cadetblue;
}

.container {
    padding: 2px 16px;
}