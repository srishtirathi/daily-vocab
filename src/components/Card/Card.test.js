/* eslint-disable react/jsx-props-no-spreading */
import { render, screen } from '@testing-library/react';
import React from 'react';
import {
  BrowserRouter,
} from 'react-router-dom';
import Card from './Card';

describe(Card.name, () => {
  const mockProps = {
    vocab: {
      word: 'hi',
      sentence: 'hii srishti',
    },

  };
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('sanity check', () => {
    expect(1 + 1).toBe(2);
  });
  it('should match snapshot', () => {
    const { container } = render(<BrowserRouter><Card {...mockProps} /></BrowserRouter>);
    expect(container).toMatchSnapshot();
  });
  it('should display name ,word and sentence', () => {
    render(<BrowserRouter><Card {...mockProps} /></BrowserRouter>);
    screen.getByText('hi');
    screen.getByText('hii srishti');
  });
});
