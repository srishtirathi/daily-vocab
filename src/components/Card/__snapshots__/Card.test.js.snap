// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<div>
  <div
    class="card"
    data-testid="card"
  >
    <div
      class="container"
    >
      <h4>
        <b>
          hi
        </b>
      </h4>
      <p>
        hii srishti
      </p>
    </div>
  </div>
</div>
`;
